src/bin/rav1e-ch.rs

```diff
@@ -0,0 +1,574 @@
+          metrics_enabled,
+        );
+
+        if verbose != Verbose::Quiet {
+          progress.add_frame(summary.clone());
+          if verbose == Verbose::Verbose {
+            info!("{} - {}", summary, progress);
+          } else {
+            // Print a one-line progress indicator that overrides itself with every update
+            eprint!("\r{}                    ", progress);
+          };
+        }
+      }
+
+      if verbose != Verbose::Quiet {
+        if verbose == Verbose::Verbose {
+          // Clear out the temporary progress indicator
+          eprint!("\r");
+        }
+        progress.print_summary(verbose == Verbose::Verbose);
+      }
+
+      // receive_packet.result()
+      Ok(())
+    });
+
+    a.join().unwrap()?;
+    b.join().unwrap()?;
+    c.join().unwrap()?;
+    d.join().unwrap()?;
+
+    Ok(())
+  })
+  .unwrap()
+}
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+  #[cfg(feature = "tracing")]
+  use rust_hawktracer::*;
+  init_logger();
+
+  #[cfg(feature = "tracing")]
+  let instance = HawktracerInstance::new();
+  #[cfg(feature = "tracing")]
+  let _listener = instance.create_listener(HawktracerListenerType::ToFile {
+    file_path: "trace.bin".into(),
+    buffer_size: 4096,
+  });
+
+  run().map_err(|e| {
+    error::print_error(&e);
+    Box::new(e) as Box<dyn std::error::Error>
+  })
+}
+
+fn init_logger() {
+  use std::str::FromStr;
+  fn level_colored(l: log::Level) -> console::StyledObject<&'static str> {
+    use console::style;
+    use log::Level;
+    match l {
+      Level::Trace => style("??").dim(),
+      Level::Debug => style("? ").dim(),
+      Level::Info => style("> ").green(),
+      Level::Warn => style("! ").yellow(),
+      Level::Error => style("!!").red(),
+    }
+  }
+
+  // this can be changed to flatten
+  let level = std::env::var("RAV1E_LOG")
+    .ok()
+    .map(|l| log::LevelFilter::from_str(&l).ok())
+    .unwrap_or(Some(log::LevelFilter::Info))
+    .unwrap();
+
+  fern::Dispatch::new()
+    .format(move |out, message, record| {
+      out.finish(format_args!(
+        "{level} {message}",
+        level = level_colored(record.level()),
+        message = message,
+      ));
+    })
+    // set the default log level. to filter out verbose log messages from dependencies, set
+    // this to Warn and overwrite the log level for your crate.
+    .level(log::LevelFilter::Warn)
+    // change log levels for individual modules. Note: This looks for the record's target
+    // field which defaults to the module path but can be overwritten with the `target`
+    // parameter:
+    // `info!(target="special_target", "This log message is about special_target");`
+    .level_for("rav1e", level)
+    // output to stdout
+    .chain(std::io::stderr())
+    .apply()
+    .unwrap();
+}
+
+cfg_if::cfg_if! {
+  if #[cfg(any(target_os = "windows", target_arch = "wasm32"))] {
+    fn print_rusage() {
+      eprintln!("Windows benchmarking is not supported currently.");
```

Comment:
Maybe add sth about wasm in the Error message also?

Reply:
Unrelated to this patch, but it could be added to the rav1e cli on master as well.